web-ng/src/app/components/layer-dialog/layer-dialog.component.ts

```diff
@@ -50,54 +47,46 @@ export class LayerDialogComponent implements OnDestroy {
   layer?: Layer;
   layerName!: string;
   projectId?: string;
-  activeProject$: Observable<Project>;
   subscription: Subscription = new Subscription();
   fieldTypes = FieldType;
   fields: List<Field>;
 
   constructor(
-    @Inject(MAT_DIALOG_DATA) data: { layerId: string },
+    @Inject(MAT_DIALOG_DATA)
+    data: {
+      layerId: string;
+      projectId: string;
+      layers: Map<string, Layer>;
```

Comment:
Why does the layer dialog need access to all the layers? Can we just pass the layer that's being edited (optional)? See https://en.wikipedia.org/wiki/Principle_of_least_privilege